concent_api/gatekeeper/views.py

```diff
@@ -353,6 +353,39 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
             concent_client_public_key
         )
 
+    # Validate category in FileInfo
+    if not all('category' in file for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response(
+            "Missing 'category' field in FileInfo.",
+            ErrorCode.MESSAGE_FILES_CATEGORY_MISSING,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            concent_client_public_key
+        )
+
+    if not all(
+        isinstance(file['category'], FileTransferToken.FileInfo.Category) for file in loaded_golem_message.files
+    ):
+        return gatekeeper_access_denied_response(
+            "'category' field has invalid value.",
+            ErrorCode.MESSAGE_FILES_CATEGORY_INVALID,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            concent_client_public_key
+        )
+
+    if not (
```

Comment:
this validation is incorrect: 
for example, if you change the test `test_upload_should_return_401_if_file_category_is_not_unique` in a way that it contains files with two different categories, the test will still pass.

Please change it to sth like:
```python
categories = [file_info['category'] for file_info in loaded_golem_message.files]
    if len(set(categories)) != len(categories):
```

Reply:
> if you change the test test_upload_should_return_401_if_file_category_is_not_unique in a way that it contains files with two different categories, the test will still pass.

That is correct behavior.

Reply:
what I ment is:
```python
    def test_upload_should_return_401_if_file_categories_are_not_unique_across_file_info_list(self):
        file_1 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.results
        )
        file_2 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-28-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.results
        )
```
produces: `MESSAGE_FILES_CATEGORY_NOT_UNIQUE`,
but if you change it to:
```python
    def test_upload_should_return_401_if_file_categories_are_not_unique_across_file_info_list(self):
        file_1 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.results
        )
        file_2 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-28-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.resources
        )
```
(so now `FileTransferToken` has files belonging to two **different** categories), you still get:
`MESSAGE_FILES_CATEGORY_NOT_UNIQUE`. Please check it and fix validation

Reply:
:+1: done